src/unix/inc/mincom/ccomptrs.h

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// ccomptrs.h base types for managing basic reference counting and 
+// casting of IUnknown types useful for when the entire platform
+// abstraction layer, or Abstract Template Libraries are not available.
+
+#pragma once
+
+template <class T>
+class _NoAddRefReleaseOnCComPtr : 
+    public T
+{
+    private:
+    
+    ~_NoAddRefReleaseOnCComPtr() {}
+    
+    private:
+        STDMETHOD_(ULONG, AddRef)()=0;
+        STDMETHOD_(ULONG, Release)()=0;
+};
+
+
+//CComPtrBase provides the basis for all other smart pointers
+//The other smartpointers add their own constructors and operators
+template <class T>
+class CComPtrBase
+{
+protected:
+	CComPtrBase()
+	{
+		p = NULL;
+	}
+	CComPtrBase(int nNull)
+	{
+		(void)nNull;
+		p = NULL;
+	}
+	CComPtrBase(T* lp)
+	{
+		p = lp;
+		if (p != NULL)
+			p->AddRef();
+	}
+public:
+	typedef T _PtrClass;
```

Comment:
>_PtrClass [](start = 11, length = 9)

not used?

Reply:
It's a public type. It's a method of getting the type associated with the template. For example:

```CComPtr<CMyType>::_PtrClass obj;```

Will generate a local variable of type CMyType.

It is part of the ATL definition of this class, so I think I'll keep it.